Add min_impute and nan warning

grromrell opened the pull request:
Fixing #421 

Just adds some runtime warnings to the DataFrameImputer. I also added a potential change to the X.fillna that could be used to limit the amount of imputing done by using the limit keyword. I thought it was worth considering as an option. Also not married to the value for max_impute, thoughts there are helpful.

Aylr commented on healthcareai/common/transformers.py:
```diff
@@ -20,11 +21,12 @@ class DataFrameImputer(TransformerMixin):
     """
 
-    def __init__(self, impute=True, verbose=True):
+    def __init__(self, impute=True, verbose=True, max_impute=.5):
```
- Great idea. Would you mind keeping the `verbose` arg last?
- At this point I think that this method needs a docstring. We prefer google format, and if you're using an editor like pycharm it's trivial to get that formatting right. No need to go overboard, just note what the arguments mean.

Aylr commented on healthcareai/common/transformers.py:
```diff
@@ -54,7 +56,17 @@ def transform(self, X, y=None):
+        for c in X:
+            pct_impute = X[c].isnull().sum() / len(X)
+            if pct_impute > self.max_impute:
+                warnings.warn("{0:.2f}% of data for column '{1}' is missing. Imputed "
```
This might read a little better if you state the column name first, then the %missing. I totally love this, and I'm going to steal your idea of using warnings for a few other thigns!

Aylr commented on healthcareai/common/transformers.py:
```diff
@@ -54,7 +56,17 @@ def transform(self, X, y=None):
+                               RuntimeWarning)
+        
+        #Alternative fill, only fill maximum number of values based on max_impute
+        #result = X.fillna(self.fill, limit=len(X)//(1/self.max_impute))
```
I'm not sure how this will affect downstream algorithms. Do you have any sense for this? If not, I'd say create another issue and remove this commented code from this PR.

Aylr commented on healthcareai/common/transformers.py:
```diff
@@ -54,7 +56,17 @@ def transform(self, X, y=None):
             return X
-
+        
+        #Warn users if %nan is too high
```
I'd be thrilled if this were factored out to a separate function with tests (and I'm happy to guide you through that if you'd like).

grromrell commented:
I added some docs, changed the way the warning prints and removed the alternative fill method. I did not separate the warn_nan into its own function as it seems like this is the only place it would get used and adding a unit test didn't seem worth it for a simple warning.
